mineral-ui/mineral-ui
# test(happo): Update to happo v1.0

### Description

Recent updates on happo.io and the happo.io client has made the CI
integration a bit simpler. Instead of having to write your own CI
script, ready-made scripts are included in the happo.io npm client. The
one for Travis, `happo-ci-travis` [1] is the one we're going to use. It
is mostly just a straight up copy-paste of what we used to have in
scripts/happo-ci.sh, with a few adjustments to make it more generic and
a little more robust.

The comment posting is no longer needed. Happo will automatically post a
build status back to the PR. If there are diffs, Happo will show a
"failure". If there are no diffs, the Happo status is green. In order to
turn a failed Happo status into a success, you go to the
details/comparison page and hit the (newly added) success button at the
top. Happo will then post back a green status to the PR.

With the new status, there's no longer a need for the comment posted to
the PR. So I removed that bit of code, including the dependency on the
`github` client (which is now `@octokit/rest` by the way).

The pre-release/rc version will be updated to a proper release as soon
as I've seen this script in action a bit more. I'll push a separate PR
following up with that change later.

[1] https://github.com/enduire/happo.io/blob/master/bin/happo-ci-travis

### Motivation and context

We no longer have to maintain our own complicated happo scripts, including posting back status to the PR. 

### Screenshots, videos, or demo, if appropriate

Happo status on a PR on github.com:
<img width="424" alt="Happo status on a PR on github.com" src="https://user-images.githubusercontent.com/578277/37517722-9bc904c6-2912-11e8-8052-357579077857.png">

Happo status inside happo.io:
<img width="786" alt="Happo status inside happo.io" src="https://user-images.githubusercontent.com/578277/37517833-fe8f3e68-2912-11e8-9e8d-5c75d20596aa.png">

### How to test

Since happo tests aren't run on outside PRs, I think it might be best for someone with push access to convert it to a PR posted from inside the project. That way, we can make sure things are okay before merging. 

### Types of changes

Mainly a refactor.

### How does this PR make you feel?

![Sweden qualifies for the World Cup](https://media.giphy.com/media/26u4cr0ttdswXKgNy/giphy.gif)
